From 97529f15f4bcb5ee82f71c6a2e212094ebf1dae3 Mon Sep 17 00:00:00 2001
From: Michel Pollet <buserror@gmail.com>
Date: Wed, 28 Nov 2012 16:07:29 +0000
Subject: [PATCH 07/11] imx23-olinuxino: Add support for hard SPI

Adds support for the SSP2 hardware driver
onto the UEXT connector of the olinuxino.
Also instantiates a spidev driver to allow
access to /dev/spidev0.0.

Signed-off-by: Michel Pollet <buserror@gmail.com>
---
 arch/arm/boot/dts/imx23-olinuxino.dts |   42 +++++++++++++++++++++++++++++++++
 1 file changed, 42 insertions(+)

diff --git a/arch/arm/boot/dts/imx23-olinuxino.dts b/arch/arm/boot/dts/imx23-olinuxino.dts
index 0a91545..e5b080d 100644
--- a/arch/arm/boot/dts/imx23-olinuxino.dts
+++ b/arch/arm/boot/dts/imx23-olinuxino.dts
@@ -24,6 +24,48 @@
 
 	apb@80000000 {
 		apbh@80000000 {
+			ssp1: ssp@80034000 {
+				compatible = "fsl,imx23-spi";
+				clock-frequency = <160000000>;
+				bus-num = <0>;
+				pinctrl-names = "default";
+				pinctrl-0 = <&spi1_control_pins &spi1_cs_pins>;
+				status = "okay";
+
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				device@0 {
+					compatible = "spidev";
+					reg = <0>;
+					spi-max-frequency = <16000000>;
+					interrupts = <&gpio2 28>;
+					interrupt-names = "spidev-trigger";
+				};
+			};
+			pinctrl@80018000 {
+				spi1_control_pins: spi1_control_pins {
+					reg = <0>;
+					fsl,pinmux-ids = <
+						0x0182 // MX23_PAD_GPMI_WRN__SSP2_SCK
+						0x0142 // MX23_PAD_GPMI_RDY1__SSP2_CMD aka MOSI
+						0x0002 // MX23_PAD_GPMI_D00__SSP2_DATA0 aka MISO
+					>;
+				};
+				spi1_cs_pins: spi1_cs_pins {
+					reg = <0>;
+					fsl,pinmux-ids = <
+						0x1033  // MX23_PAD_LCD_D03__GPIO_1_3 
+						// This is optional, this allows the spidev
+						// driver to have an interrupt pin, this is
+						// "Bank 2 pin 28" and it needs a pullup
+						//
+						0x21c3 // MX23_PAD_GPMI_CE0N__GPIO_2_28
+					>;
+					fsl,pull-up = <1>;
+				};
+			};
+			
 			ssp0: ssp@80010000 {
 				compatible = "fsl,imx23-mmc";
 				pinctrl-names = "default";
-- 
1.7.10.4

